tests/test_utils.ts

```diff
@@ -137,3 +137,24 @@ async function spawnServer(
 
   return { serverProcess, lines, address };
 }
+
+/**
+ * Wait until the passed function returns a truthy value or the time runs out
+ */
+export async function waitFor(
```

Comment:
Perhaps, you could use https://deno.land/std@0.195.0/async/mod.ts?s=retry

It's default values look like they might suffice here.

Reply:
Good point, I really should look more as to what's available in the standard library.